x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/RollupShardIndexer.java

```diff
@@ -90,11 +92,15 @@ class RollupShardIndexer {
     private final String[] metricFields;
     private final String[] labelFields;
     private final Map<String, FieldValueFetcher> fieldValueFetchers;
+    private final AtomicLong numReceived = new AtomicLong();
```

Comment:
Maybe these counters should be moved to the  `RollupShardTask` class? If this class has access to the task then it can just increments these counters that are part of the task. I think makes managing these counters easier.